src/main/java/com/moneylendingapp/controllers/AuthController.java

```diff
@@ -0,0 +1,24 @@
+package com.moneylendingapp.controllers;
+
+import com.moneylendingapp.dto.requests.SignUpRequest;
+import com.moneylendingapp.dto.responses.SignUpResponse;
+import com.moneylendingapp.services.UserService;
+import lombok.RequiredArgsConstructor;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+
+@RestController
+@RequestMapping("/api/v1/auth")
+@RequiredArgsConstructor
+public class AuthController {
+
+    private final UserService userService;
+
+    @PostMapping("/signup")
+    public ResponseEntity<SignUpResponse> registerUser(@Validated @RequestBody SignUpRequest signUp) {
```

Comment:
Create a ResponseAdvice to wrap your response with responseEntity.
or use @ResponseStatus status to specify the exact response status